### ISSUE TITLE
How do you avoid "Possibly unhandled error" exceptions?

### ISSUE THREAD

**Issue by fresheneesz** [petkaantonov/bluebird#665]
I have this code:

```
var x = Promise.try(function() {
    throw new Error("error")
})

setTimeout(function (){
    x.catch(function() {
        console.log("caught")
    })
})
```

`x` eventually catches it as it should, but how do I tell bluebird not to output a "possibly unhandled error" exception for x?

**Comment by renedx** [comment]
You're able to overwrite it, if you can't catch them properly?
https://github.com/petkaantonov/bluebird#error-handling

**Comment by benjamingr** [comment]
Can you show code you'd actually write that does this?

In my experience code that attaches a catch handler asynchronously isn't really a thing in production code

**Comment by fresheneesz** [comment]
@renedx Yes, I suppose you could do

```
var ignoreThese = []
Promise.onPossiblyUnhandledRejection(function(e, promise) {
    if(!ignoreThese.indexOf(ignoreThese))
      throw e;
});

var x = ...;
ignoreThese.push(x)
...
```

Doesn't seem like the ideal way to handle it. I'd rather something like `x.detach()` to do this automatically.

@benjamingr Less contrived use case - preloading:

```
var x = somethingThatTakesAWhile()
// later ...
somePromiseChain.then(function() {
   return x
}).then(function() {
   // use x
}).catch(function(e) {
  
})

```

**Comment by benjamingr** [comment]
I'm not sure how that's any less contrived - you just have `// later` there. Why are you not attaching a handler? 

> var x = somethingThatTakesAWhile()

If you want to opt out for that specifically it'd be

```js
var x = somethingThatTakesAWhile()
x.catch(() => {}); // add empty catch
```

Or with 3.0

```js
x.suppressUnhandledRejections()
```

**Comment by fresheneesz** [comment]
@benjamingr Ohhhh, that's the answer i was looking for: `x.suppressUnhandledRejections()`, tho I guess adding an empty catch has the exact same effect, right? 

But as far as my example, just using pure `then`s won't help me, but I could use `all` to help me there:

```
var x = somethingThatTakesAWhile()
Promise.all([x, somePromiseChain]).then(function(x, y) {
   // use x
}).catch(function(e) {
   // ...
})
```

Not used to bluebird's utilities yet. But you answered my question anyway, so I'll close this up!

**Comment by benjamingr** [comment]
> tho I guess adding an empty catch has the exact same effect too, huh?

Not with 3.0 cancellation. 

> But as far as my example, just using pure thens won't help me, but I could use all to help me there:

Yes, generally these things where people add `.catch` asynchronously tend to coincide with race conditions which is why I always asked.

Glad I could help. In general it's best to ask support questions in Stack Overflow (with the bluebird tag) in the future and keep the issue tracker for feature requests and bugs.

**Comment by fresheneesz** [comment]
Mk, will do - sometimes hard to know if a github module's contributors watch for SO questions
